File: 01/one.clj
(require '[clojure.string :as str])

(defn calc
  "Sum every frequency shift to get the resulting frequency."
  [shifts]
  (reduce + (map #(Long/parseLong %) shifts)))

(let [shifts (->> (slurp "input.txt")
                  str/split-lines
                  (remove str/blank?))]
  (println (calc shifts)))

File: 01/one.exs
defmodule Day01.One do
  def calc(input_data) do
    input_data
    |> Enum.map(&String.to_integer/1)
    |> Enum.sum()
  end
end

"input.txt"
|> File.read!()
|> String.split("\n", trim: true)
|> Day01.One.calc()
|> IO.puts()
